lib/client.ts

```diff
@@ -597,13 +602,10 @@ export class ServiceClient {
       randomize
     };
 
-    const operation = retry.operation(opts);
     const retryErrors: ServiceClientError[] = [];
-
-    const breaker = this.getCircuitBreaker(params);
-
-    return new Promise<ServiceClientResponse>((resolve, reject) =>
-      operation.attempt((currentAttempt: number) => {
+    return new Promise<ServiceClientResponse>((resolve, reject) => {
+      const breaker = this.getCircuitBreaker(params);
+      const retryOperation = operation(opts, (currentAttempt: number) => {
```

Comment:
I am not exactly sure that the simplifications of this file make it worth it starting to fork and maintain a library as part of Perron.

Reply:
Updated the description to make it clearer.